src/HttpRequestHandler.cpp

```diff
@@ -38,52 +41,55 @@ HttpResponse *HttpRequestHandler::parseErrorResponse(Server *server, std::pair<u
 
 void HttpRequestHandler::handleRequest(Client *client, Server *server)
 {
-	switch (client->getRequest()->getMethod())
+	try
 	{
-		case Util::METHOD::GET:
+		switch (client->getRequest()->getMethod())
 		{
-			parseOkResponse(client, server);
-			return;
-		}
+			case Util::METHOD::GET:
+			{
+				parseOkResponse(client, server);
+				return;
+			}
 
-		case Util::METHOD::POST:
-		{
-			std::cout << "we would do post here" << std::endl;
-			parseOkResponse(client, server);
-			return;
-		}
+			case Util::METHOD::POST:
+			{
+				Methods::executePost(client->getRequest(), client->getServer());
+				parseOkResponse(client, server);
+				return ;
+			}
 
-		case Util::METHOD::DELETE:
-		{
-			std::cout << "we would do delete here" << std::endl;
-			parseOkResponse(client, server);
-			return;
-		}
+			case Util::METHOD::DELETE:
+			{
+				Methods::executeDelete(*client->getRequest());
+				parseOkResponse(client, server);
+				return;
+			}
 
-		case Util::METHOD::CGI_GET:
-		{
-			std::cout << "we would execute get request cgi here" << std::endl;
-			std::string cgiResponse = CgiHandler::executeCgi(*client->getRequest());
-			parseOkResponse(client, server);
-			client->getResponse()->setCgiResponse(cgiResponse);
-			return;
-		}
+			case Util::METHOD::CGI_GET:
+			{
+				std::string cgiResponse = CgiHandler::executeCgi(*client->getRequest());
+				parseOkResponse(client, server);
+				client->getResponse()->setCgiResponse(cgiResponse);
+				return;
+			}
 
-		case Util::METHOD::CGI_POST:
-		{
-			std::cout << "we would execute post request cgi here" << std::endl;
-			std::string cgiResponse = CgiHandler::executeCgi(*client->getRequest());
-			parseOkResponse(client, server);
-			client->getResponse()->setCgiResponse(cgiResponse);
-			return;
-		}
+			case Util::METHOD::CGI_POST:
+			{
+				std::string cgiResponse = CgiHandler::executeCgi(*client->getRequest());
+				parseOkResponse(client, server);
+				client->getResponse()->setCgiResponse(cgiResponse);
+				return;
+			}
 
-		default :
-			throw MethodNotAllowedException("Method not allowed");
-			break;
+			default :
+				throw MethodNotAllowedException("Method not allowed");
+				break;
+		}
+	}
+	catch (const Exception& e)
+	{
+		client->setResponse(parseErrorResponse(server, ExceptionManager::getErrorStatus(e)));
```

Comment:
just a quick thing, not need to change this but setting the response in the catch block is essentially a different thing than to do it completely outside this try catch.
-> To set the response in the throw block gets executed only in the case we throw an error somewhere in the try block.
-> To set the error response after the switch case and try catch means that if for some reason we would come out of the switch case (which we should not do, but you never know), we know for sure that something weird happened as we arrived here instead of returning early -> we respond with internal server error to the client like "okay something weird is going on here but here you go client".

Reply:
aka I would still add this error seponse 500 after the try catch block in the end of this whole function as a fail safe measure